src/solvers/dgsem_tree/indicators_1d.jl

```diff
@@ -8,19 +8,22 @@
 # this method is used when the indicator is constructed as for shock-capturing volume integrals
 function create_cache(::Type{IndicatorHennemannGassner},
                       equations::AbstractEquations{1}, basis::LobattoLegendreBasis)
-    alpha = Vector{real(basis)}()
+    uEltype = real(basis)
+    alpha = Vector{uEltype}()
     alpha_tmp = similar(alpha)
 
-    A = Array{real(basis), ndims(equations)}
-    indicator_threaded = [A(undef, nnodes(basis)) for _ in 1:Threads.maxthreadid()]
-    modal_threaded = [A(undef, nnodes(basis)) for _ in 1:Threads.maxthreadid()]
+    MA1d = MArray{Tuple{nnodes(basis)},
+                  uEltype, 1, nnodes(basis)}
+
+    indicator_threaded = MA1d[MA1d(undef) for _ in 1:Threads.maxthreadid()]
+    modal_threaded = MA1d[MA1d(undef) for _ in 1:Threads.maxthreadid()]
```

Comment:
Why did you change the type from a standard `Array` to an `MArray`? (The same question applies to a lot of places below)

Reply:
The compiler may generate more efficient code using the additional information from StaticArrays.jl, but the compile time (latency) typically increases, too. If you change any types here, please benchmark the runtime performance and latency (time to first simulation) for increasing polynomial degrees (from 2 to 7 or 15).

I agree that the current mess is not optimal, but I doubt that four-dimensional arrays will benefit from StaticArrays.jl.

Reply:
Following the rule of thumb for static arrays that these are [best applicable for arrays with number of entries <= 100](https://juliaarrays.github.io/StaticArrays.jl/stable/#When-Static-Arrays-may-be-useful), it probably makes sense to switch at least 1D and 2D to statically sized. If we keep 3d as is, we should at least add a comment there. That being said, only for polydeg = 4 and above we have in this case more than 100 entries (k=3 still has only 64 entries).